app/components/doi-related-identifier.js

```diff
@@ -0,0 +1,213 @@
+import Component from '@ember/component';
+import { inject as service } from '@ember/service';
+
+const relationTypeList = [
+  'IsCitedBy',
+  'Cites',
+  'IsSupplementTo',
+  'IsSupplementedBy',
+  'IsContinuedBy',
+  'Continues',
+  'IsDescribedBy',
+  'Describes',
+  'HasMetadata',
+  'IsMetadataFor',
+  'HasVersion',
+  'IsVersionOf',
+  'IsNewVersionOf',
+  'IsPreviousVersionOf',
+  'IsPartOf',
+  'HasPart',
+  'IsReferencedBy',
+  'References',
+  'IsDocumentedBy',
+  'Documents',
+  'IsCompiledBy',
+  'Compiles',
+  'IsVariantFormOf',
+  'IsOriginalFormOf',
+  'IsIdenticalTo',
+  'IsReviewedBy',
+  'Reviews',
+  'IsDerivedFrom',
+  'IsSourceOf',
+  'IsRequiredBy',
+  'Requires',
+  'IsObsoletedBy',
+  'Obsoletes',
+];
+
+const relatedIdentifierTypeList = [
+  'ARK',
+  'arXiv',
+  'bibcode',
+  'DOI',
+  'EAN13',
+  'EISSN',
+  'Handle',
+  'IGSN',
+  'ISBN',
+  'ISSN',
+  'ISTC',
+  'LISSN',
+  'LSID',
+  'PMID',
+  'PURL',
+  'UPC',
+  'URL',
+  'URN',
+  'w3id',
+];
+
+const resourceTypeGeneralList = [
+  'Audiovisual',
+  'Collection',
+  'DataPaper',
+  'Dataset',
+  'Event',
+  'Image',
+  'InteractiveResource',
+  'Model',
+  'PhysicalObject',
+  'Service',
+  'Software',
+  'Sound',
+  'Text',
+  'Workflow',
+  'Other',
+];
+
+export default Component.extend({
+  store: service(),
+  relationTypeList,
+  relationTypes: relationTypeList,
+  relatedIdentifierTypeList,
+  relatedIdentifierTypes: relatedIdentifierTypeList,
+  controlledIdentifierType: false,
+  isMetadataRelationType: false,
+  isMetadataRelationTypes: [ 'HasMetadata', 'IsMetadataFor' ],
+  resourceTypeGeneralList,
+  resourceTypesGeneral: resourceTypeGeneralList,
+
+  didReceiveAttrs() {
+    this._super(...arguments);
+
+    if (relatedIdentifierTypeList.includes(this.fragment.get('relatedIdentifierType'))) {
+      this.set('controlledIdentifierType', true);
+    } else {
+      this.set('controlledIdentifierType', false);
+    }
+    if (this.isMetadataRelationTypes.includes(this.fragment.get('relationType'))) {
+      this.set('isMetadataRelationType', true);
+    } else {
+      this.set('isMetadataRelationType', false);
+    }
+  },
+  updateRelatedIdentifier(value) {
+    const ark = /^ark:\/[0-9]{5}\/\S+$/;
+    const lsid = /^[uU][rR][nN]:[lL][sS][iI][dD]:(A-Za-z0-9][A-Za-z0-9()+,-.=@;$_!*'"%]):(A-Za-z0-9][A-Za-z0-9()+,-.=@;$_!*'"%]):(A-Za-z0-9][A-Za-z0-9()+,-.=@;$_!*'"%])[:]?(A-Za-z0-9][A-Za-z0-9()+,-.=@;$_!*'"%])?$/;
+    const purl = /^http?:\/\/(purl\.oclc\.org\/)/;
+    const isbn = /^(?:ISBN(?:-10)?:?●)?(?=[0-9X]{10}$|(?=(?:[0-9]+[-●]){3})[-●0-9X]{13}$)↵[0-9]{1,5}[-●]?[0-9]+[-●]?[0-9]+[-●]?[0-9X]$/;
+    const arxiv = /^(arXiv:)(\d{4}.\d{4,5}|[a-z\-]+(\.[A-Z]{2})?\/\d{7})(v\d+)?/;
+    const doi = /^(10\.\d{4,5}\/.+)/;
+    const bibcode = /\d{4}[A-Za-z\.\&]{5}[\w\.]{4}[ELPQ-Z\.][\d\.]{4}[A-Z]/;
+    const url = /^(http|https)?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&//=]*)/;
```

Comment:
Please look into the npm `validator` package that we are already using to make sure this is consistent, e.g. for URL. 

Why is it possible to set a relatedIdentifierType not in the controlled list?

Reply:
it is not possible. The validation will always require one and the select will only let you choose from the list. the select doesn't has the ability to  add new types

Reply:
Please see app/components/doi-related-identifier.js:174.